test/parallel/test-no-enter-tickcallback.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+require('../common');
+
+/**
+ * This test relies upon the following internals to work as specified:
+ *  - require('domain') causes node::Environment::set_tick_callback_function()
+ *    to use process._tickDomainCallback() to process the nextTickQueue;
+ *    replacing process._tickCallback().
+ *  - setImmediate() uses node::MakeCallback() instead of
+ *    node::AsyncWrap::MakeCallback(). Otherwise the test will always pass.
+ *    Have not found a way to verify that node::MakeCallback() is used.
+ */
```

Comment:
Could these conditions be checked by adding a test for the reverse situation? I.e. making sure that if there are ticks pending, the `_tickDomainCallback` is actually executed?